source/Calamari/Kubernetes/ResourceStatus/ResourceStatusReportWrapper.cs

```diff
@@ -92,14 +92,15 @@ public bool IsEnabled(ScriptSyntax syntax)
                 return new CommandResult(string.Empty, 1);
             }
 
-            var stabilizingTimer = new StabilizingTimer(
-                new CountdownTimer(TimeSpan.FromSeconds(deploymentTimeoutSeconds)),
-                new CountdownTimer(TimeSpan.FromSeconds(stabilizationTimeoutSeconds)));
+            var deploymentTimer = deploymentTimeoutSeconds == 0
+                ? new InfiniteCountdownTimer() as ICountdownTimer
+                : new CountdownTimer(TimeSpan.FromSeconds(deploymentTimeoutSeconds)) as ICountdownTimer;
```

Comment:
no need to the case ` as ICountdownTimer`

Reply:
It fails compilation for lower versions of C# on TC, like this:

```
C:\BuildAgent\work\e0cefbed4ad11812\source\Calamari\Kubernetes\ResourceStatus\ResourceStatusReportWrapper.cs(97,47): error CS8957: Conditional expression is not valid in language version 7.3 because a common type was not found between 'InfiniteCountdownTimer' and 'CountdownTimer'. To use a target-typed conversion, upgrade to language version 9.0 or greater. [C:\BuildAgent\work\e0cefbed4ad11812\source\Calamari\Calamari.csproj]
```

Reply:
Wondering if we could explicitly declare the type instead fo `var`?

Reply:
I tried but that didn't work either, because the right hand side of the assignment is evaluated first and the error occurred there. Without the "target-typed conversion" feature,  the type we declared on the left hand side didn't take effect until the right-hand side is evaluated.